ext/opcache/ZendAccelerator.c

```diff
@@ -2425,6 +2486,152 @@ int accel_post_deactivate(void)
 		return SUCCESS;
 	}
 
+	/* Free static variables of preloaded classes and functions */
+	if (ZCSG(preload_script)) {
+		uint32_t count;
+		Bucket *p;
+
+		if (ZCSG(preload_script)->script.function_table.nNumUsed) {
+			count = ZCSG(preload_script)->script.function_table.nNumUsed;
+			p = EG(function_table)->arData + (EG(function_table)->nNumUsed - count);
+
+			do {
+				zend_op_array *op_array = Z_PTR(p->val);
+
+				if (op_array->type == ZEND_USER_FUNCTION) {
+					ZEND_ASSERT(op_array->fn_flags & ZEND_ACC_IMMUTABLE);
+					if (op_array->static_variables) {
+						destroy_op_array(op_array);
```

Comment:
Should this be freeling the op_array rather than just the static_variables table in the request_data?

Reply:
For "immutable" arrays we have to free only static_variables. Everything else kept in SHM, except  for run_time cache, that is in CG(arena). Or may be I misunderstood the question.

Reply:
I mean, the code here currently frees the op_array, but the assertion above says that this is an immutable op_array, so I would expect it to only free static_variables.